src/main/java/appeng/crafting/v2/FastToIterateArrayBasedMultiMap.java

```diff
@@ -0,0 +1,258 @@
+package appeng.crafting.v2;
```

Comment:
Not sure about the package here, but since this data structure has a very specific use case it didn't put it in `util`